### PR TITLE
CPS-???? | Ada token supply exhaustion

### PR DESCRIPTION
**Pull request by xezon**
Hello. I hope this is not a duplicate and helps. At a glance I could not find a CPS for it.

Keep up the good work!

---

([rendered in PR branch](https://github.com/xezon/cardano-CIPs/blob/cps-ada-token-supply-exhaustion/CPS-XXXXX/README.md))

### CODE REVIEW

File: CPS-XXXXX/README.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
+---
+CPS: ?
+Title: ADA token supply exhaustion
+Status: Open
+Category: Token
```
Comment (rphair) [inline CPS-XXXXX/README.md:5]:
```suggestion
Category: Ledger
```

**Review by rphair — COMMENTED** [review]
Pending full consideration of this (myself I haven't read every detail yet), I'm correcting this to category `Ledger` because one way or another it proposes changing its size or granularity:

File: CPS-XXXXX/README.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
+---
+CPS: ?
+Title: ADA token supply exhaustion
+Status: Open
+Category: Ledger
+Authors:
+  - xezon
+Proposed Solutions:
+Discussions:
```
Comment (Ryun1) [inline CPS-XXXXX/README.md:9]:
```suggestion
Discussions:
  - https://github.com/cardano-foundation/CIPs/pull/630/
```

**Comment by rphair** [comment]
@Ryun1 @xezon while on the matter of discussions: this isn't the first time this issue has come up on the Cardano Forum, mostly about the 2 issues of fractionalization / granularity and "lost ADA" due to misplaced keys including deceased holders, e.g.: https://forum.cardano.org/t/cardano-ada-token-supply/238/6

Because these are all social issues I am still not certain this is a matter for the CIP process, but not saying for sure until I can read this in greater detail for the part of the submission that is purely technical.  I'm looking forward to giving this a fair & complete read in this light (before we build agenda the CIP meeting on 12 December, for sure).

**Review by rphair — COMMENTED** [review]
p.s. @xezon apologies for my last comment offering to leave you hanging for a week after an initially negative response.  So I cleared time to give it a complete read & response and, although I do think it is _not appropriate_ as a CPS, I wanted to give my own reasons in advance & put it on the next CIP Editors' meeting agenda so we can decide by consensus with other editors & community members: https://hackmd.io/@cip-editors/78

You are right this is a "duplicate" in the sense that this issue comes up repeatedly: though never as a CIP or CPS before (mainly on CF Forum and probably Reddit).  It also would have long been debated at IOG before deciding on an immutable fixed token supply of 45 quadrillion indivisible units.

I am going to take a stand that difficulties of this fixed supply are social and non-technical.  But beyond that, the 500-year projections of population and wealth distribution ignore the most basic features of society and history which tell us repeatedly that something else about the environment will inevitably change long before these horizons: making our original projections irrelevant.  

As the Devil's Advocate against this proposal I am (with apology) going to be a bit harsh, for purposes of debate: Having this as a CPS would legitimatise the maximalism notorious in cryptocurrency communities and invite a well-deserved bit of ridicule, since these constraints would only present difficulties if Cardano became the "one true cryptocurrency" for decades, eclipsing not only all other cryptos but also every other medium of exchange.

But long before that, other economic structures will develop even in the new months & years which will dwarf any native cryptocurrency's supply & granularity in significance: e.g. use as collateral (for more flexible & granular transactions elsewhere, even off-chain), or new structures like evolving NFTs which contain and denominate economic value some other way.

I believe this material is useful as an economic hypothesis... useful as a reference point, but not a practical matter... and should stand in a public place but _not in the CIP archive_.  Both CIPs and CPSs are about practical solutions with technical justifications, and until the community cries out over the value of 1 lovelace becoming too significant, we do not have that justification: and because of the chaotic nature of society & economy I am betting my own livelihood that we never will.

In any case I'm copying the past & present architects of the Ledger to see if they have anything to add either about the finer divisibility of Ada or what any added precision would do to the currently overwhelming Ledger calculations (@JaredCorduan @WhatisRT).  There is only one CIP that could derive from this CPS (another reason why invalid)... to further fractionalise Ada... so if these ideas belong anywhere it should be as such a CIP with the Ledger team's endorsement in advance (see [CIP-0084](https://github.com/cardano-foundation/CIPs/blob/master/CIP-0084/README.md)).

While awaiting the next CIP meeting I look forward to others' assessments & will plan to summarise the "for" and "against" points of view at that meeting.

**Comment by xezon** [comment]
Thank you for taking a look. You are likely right that in our lifetime this is no problem. But it is a problem. If there is interest to explore this further, I can offer to run a bunch of simulations with varying inputs and randomizations to render some outcomes. Perhaps that would paint a better picture to how close or far the exhaustion point is.

Apologies if this was not the correct place to bring this forward. I do not engage in forums, nor am I working for IOG.

> Having this as a CPS would legitimatise the maximalism notorious in cryptocurrency communities and invite a well-deserved bit of ridicule, since these constraints would only present difficulties if Cardano became the "one true cryptocurrency" for decades, eclipsing not only all other cryptos but also every other medium of exchange.

I am not involved enough to know about this problem. That reads like a social issue for sure. I am just concerned about function.

**Comment by Ryun1** [comment]
Hey @xezon!

I have put this proposal on the Triage agenda for CIP Editors Call # 78 (December 12th 4pm UTC), held in [The CIP Editors Meetings Discord](https://discord.gg/jvgrKbER), see [agenda](https://hackmd.io/@cip-editors/78) and [discord event](https://discord.com/events/971785110770831360/1179107190796210316).

Would be great to see you there to discuss this proposal.

File: CPS-XXXXX/README.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
+---
+CPS: ?
+Title: ADA token supply exhaustion
```
Comment (rphair) [inline CPS-XXXXX/README.md:3]:
```suggestion
Title: Ada token supply exhaustion
```
I've corrected the PR title according to official capitalisation (`ADA` is the trading symbol, not the token): https://cardano.org/what-is-ada

**Comment by xezon** [comment]
> Would be great to see you there to discuss this proposal.

Ok I will try to be there.

**Review by rphair — COMMENTED** [review]
To summarise the dialogue at today's CIP meeting: The author has offered to produce a finer grained simulation of how requirements for distinct units of Ada supply would increase with time, but still there is doubt having this as a CPS, including doubt that ti exposes any technical matters that are not already public knowledge _and_ declared to be far beyond Cardano's usage horizon even for the next few decades.

Personally all my own reservations about approving this are already in https://github.com/cardano-foundation/CIPs/pull/630#pullrequestreview-1760973373 but I would further add that the CPS, even if desirable, would not be complete without incorporating @WhatisRT's official point of view, as current Ledger architect, in https://github.com/cardano-foundation/CIPs/pull/630#issuecomment-1838534222 about it effectively breaking Plutus.

**Comment by xezon** [comment]
I have added a simulator.py script.

At the top it contains a bunch of definite values that set up the ground rules for the simulation with Cardano's Ada.

In the run function it then sets a bunch of estimated and guessed ranged parameters to run a user population and coin units redistribution against. Once the latest population has on average less than 50,000 units to work with, the simulation stops.

## Results

Running against 10000 simulations yields the following results:

```
Ran simulation 10000 times.
Min year when min acceptable unit size (50000) was hit: 2031
Max year when min acceptable unit size (50000) was hit: 2911
Avg year when min acceptable unit size (50000) was hit: 2406
```

## Logic

For the simulation logic I arranged users into buckets, where one new bucket with users is added every year and then the coin units will trickle down from earlier buckets to newer buckets. Older buckets will eventually trickle down less coins, to try simulate the wealth gap.

The simulator currently ignores all coins of the founding members distribution, because I was not sure what to do with them. I think it is not so important.

## Parameters and ranges

I commented the parameters in the script. Hopefully that explains it good enough. Some values have data support and therefore narrower ranges, but other values have very wide ranges because of speculation. But that is ok. It is a simulation so we can just run many of them to get an average result.

## Interpretation

The result does illustrate that there is a supply exhaustion point, but also that this can be far in the future. By this simulation logic and setup we can expect it to happen in about 500 years with 50% certainty - if Cardano Ada was still used by then.

**Comment by rphair** [comment]
@xezon since according to your own calculations this "problem" scenario cannot be realised in our lifetime or the realistic lifetime of the blockchain & its token supply, we have to close this request as never having been validated as a CPS.

In the meantime we've posted recommendations about where to post such theoretical / individual material and personally I think your submission would make a good technical article and/or "thought experiment" of educational value: https://github.com/cardano-foundation/CIPs/wiki/2.-CIPs-for-Reviewers-&-Authors#when-is-a-design-document-not-a-cip

**Pull request closed without merging**